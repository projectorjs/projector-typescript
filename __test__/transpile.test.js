// @flow

const ts = require("../index");

describe("#transpile", () => {
  const code = `import {f} from "foo";
export const x = f();`;

  test("should compile simple case", () => {
    return ts.transpile({ code }).then(result => expect(result).toMatchSnapshot());
  });

  test("should respect compilerOptions", () => {
    return ts
      .transpile({ code, compilerOptions: { target: "ES2015" } })
      .then(result => expect(result).toMatchSnapshot());
  });

  test("should compile using all transpile options", () => {
    return ts
      .transpile({ code, fileName: "test.ts", moduleName: "projector-typescript" })
      .then(result => expect(result).toMatchSnapshot());
  });
});
