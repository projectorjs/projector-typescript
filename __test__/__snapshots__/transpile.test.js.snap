// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#transpile should compile simple case 1`] = `
Object {
  "diagnostics": Array [],
  "outputText": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var foo_1 = require(\\"foo\\");
exports.x = foo_1.f();
",
  "sourceMapText": undefined,
}
`;

exports[`#transpile should compile using all transpile options 1`] = `
Object {
  "diagnostics": Array [],
  "outputText": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var foo_1 = require(\\"foo\\");
exports.x = foo_1.f();
",
  "sourceMapText": undefined,
}
`;

exports[`#transpile should respect compilerOptions.target 1`] = `
Object {
  "diagnostics": Array [],
  "outputText": "import { f } from \\"foo\\";
export const x = f();
",
  "sourceMapText": undefined,
}
`;
